src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -711,7 +731,10 @@ private static Expression BindParameterFromBody(ParameterInfo parameter, bool al
         {
             if (factoryContext.JsonRequestBodyType is not null)
             {
-                throw new InvalidOperationException("Action cannot have more than one FromBody attribute.");
+                factoryContext.HasAnotherBodyParameter = true;
+#pragma warning disable CS8604 // Possible null reference argument.
+                factoryContext.TrackedParameters.Add(parameter.Name, "We failed to infer this parameter. Did you forget to inject it as a Service?");
+#pragma warning restore CS8604 // Possible null reference argument.
```

Comment:
```suggestion
                factoryContext.TrackedParameters.Add(parameter.Name!, "We failed to infer this parameter. Did you forget to inject it as a Service?");
```

Saves some lines. And we do know this is safe because we check if `if (parameter.Name is null)` at the top of `CreateArgument()`.